render/egl.c

```diff
@@ -125,19 +135,95 @@ static void init_dmabuf_formats(struct wlr_egl *egl) {
 
 bool wlr_egl_init(struct wlr_egl *egl, EGLenum platform, void *remote_display,
 		EGLint *config_attribs, EGLint visual_id) {
-	if (!load_glapi()) {
+	const char *exts_str = eglQueryString(EGL_NO_DISPLAY, EGL_EXTENSIONS);
+	if (exts_str == NULL) {
+		wlr_log(WLR_ERROR, "Failed to query EGL extensions");
 		return false;
 	}
 
-	if (eglDebugMessageControlKHR) {
+	if (!check_egl_ext(exts_str, "EGL_EXT_platform_base") ||
+			!load_egl_proc(&egl->procs.eglGetPlatformDisplayEXT,
+				"eglGetPlatformDisplayEXT") ||
+			!load_egl_proc(&egl->procs.eglCreatePlatformWindowSurfaceEXT,
+				"eglCreatePlatformWindowSurfaceEXT")) {
+		wlr_log(WLR_ERROR, "EGL_EXT_platform_base not supported");
+		return false;
+	}
+
+	if (check_egl_ext(exts_str, "EGL_KHR_image_base")) {
+		egl->exts.image_base_khr = true;
+		if (!load_egl_proc(&egl->procs.eglCreateImageKHR, "eglCreateImageKHR") ||
+				!load_egl_proc(&egl->procs.eglDestroyImageKHR, "eglDestroyImageKHR")) {
+			return false;
+		}
+	}
+
+	egl->exts.buffer_age_ext =
+		check_egl_ext(exts_str, "EGL_EXT_buffer_age");
+
+	if (check_egl_ext(exts_str, "EGL_EXT_swap_buffers_with_damage")) {
+		egl->exts.swap_buffers_with_damage_ext = true;
+		if (!load_egl_proc(&egl->procs.eglSwapBuffersWithDamageEXT,
+				"eglSwapBuffersWithDamageEXT")) {
+			return false;
+		}
+	}
+	if (check_egl_ext(exts_str, "EGL_KHR_swap_buffers_with_damage")) {
+		egl->exts.swap_buffers_with_damage_khr = true;
+		if (!load_egl_proc(&egl->procs.eglSwapBuffersWithDamageKHR,
+				"eglSwapBuffersWithDamageKHR")) {
+			return false;
+		}
+	}
```

Comment:
We don't need to save both of these. I'd check for KHR and only use EXT if that fails.
They have exactly the same function signature and semantics.

Reply:
Done

Reply:
Was this lost in a rebase or something? It's still here.

Reply:
Hmm, yeah I probably overwrote the fix during a rebase. Fixed again.